DataFormats/src/WeightedObject.cc

```diff
@@ -0,0 +1,55 @@
+#include "flashgg/DataFormats/interface/WeightedObject.h"
+
+namespace flashgg {
+
+    void WeightedObject::setWeight( string key, float val )
+    {
+        auto found_label = std::lower_bound( _labels.begin(), _labels.end(), key );
+        if( found_label == _labels.end() || *found_label != key ) {
+            _labels.push_back( key );
+            _weights.push_back( val );
```

Comment:
This is still not quite the right behavior, you need to insert the value at the location where it should be in both vectors, otherwise the sort is broken.
Needs to be:
```
_weights.insert(_weights.begin()+std::distance(_labels.begin(),found_label),val); 
_labels.insert(found_label,key);
```